backend/server/server.go

```diff
@@ -1036,3 +1040,122 @@ Click "Approve" button to apply the schema update.`,
 
 	return nil
 }
+
+func (s *Server) backfillInstanceChangeHistory(ctx context.Context) {
+	err := func() error {
+		migratedInstanceList, err := s.store.ListInstanceHavingInstanceChangeHistory(ctx)
+		if err != nil {
+			return err
+		}
+		instanceMigrated := make(map[int]bool)
+		for _, id := range migratedInstanceList {
+			instanceMigrated[id] = true
+		}
+
+		instanceList, err := s.store.ListInstancesV2(ctx, &store.FindInstanceMessage{})
+		if err != nil {
+			return err
+		}
+
+		principalList, err := s.store.GetPrincipalList(ctx)
+		if err != nil {
+			return err
+		}
+		nameToPrincipal := make(map[string]*api.Principal)
+		for _, principal := range principalList {
+			nameToPrincipal[principal.Name] = principal
+		}
+
+		var errList error
+		for _, instance := range instanceList {
+			if instance.Engine == db.Redis {
```

Comment:
This is unneeded.

Reply:
redis driver doesn't support FindMigraitonHistoryList, and we know that we don't need to migrate for redis instances. So let's just skip redis.